MaxText/llama_or_mistral_ckpt.py

```diff
@@ -228,6 +239,164 @@ def permute_to_match_maxtext_rope(arr):
   return np.concatenate((evens, odds), axis=arr.ndim - 1)
 
 
+def convert_lora_weights_to_jax_weights(lora_config, model_size):
```

Comment:
nit: this is a very long function. of course you're following the existing "style" but i would love it if someone took the initiative to improve things :D for a willing party, the for loop could be modularized and some of the structures could be pulled out to constants or something?

Reply:
I totally agree that it need to be refactored and modularized. Instead there should be a modularized library to convert any format to MaxText and vice-versa. But unfortunately it will not fit in my bandwidth/timelines.